src/change_data.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2013-2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use libc;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ChangeData<'a> {
+    UChars(&'a [u8]),
+    UShorts(&'a [u16]),
+    ULongs(&'a [libc::c_ulong]),
+    UChar(u8),
+    UShort(u16),
+    ULong(libc::c_ulong),
```

Comment:
Inner types don't need have `pub` ?

Reply:
And here no `u32` cases

Reply:
u32/guint is not mentioned in the documentation of the function

Reply:
> This seems to allow arbitrary types, I'm not sure how this function is supposed to work safely.

By limiting the number of entries. If we need more, we can always add them afterwards.

> Inner types don't need have `pub` ?

Woups... Thanks!

Reply:
Ok then. Do we know of any code actually using this binding? How do they use it?